new MCH track format + update of tracking workflows

pillot opened the pull request.

pillot committed:
new MCH track format + update of tracking workflows

aphecetche commented:
@pillot ahem, given the size of the PR, I would not mind a little PR description... please. Thanks.

pillot commented:
Well, everything is is the title. ;) There are a lot of files changed because:

1) I added a new track format TrackMCH.cxx(h) and therefore I changed the I/O of every tracking workflows, dropping the handmade message payloads to replace them by vectors of messageable objects

2) I moved all the workflows in the directory Workflow and split them to have one workflow per processor spec, which allows to chain them differently depending on what we want to run (old or new track finding, track refitting, ...)

I also added/completed 2 README, one to briefly describe the tracking algorithms and tools and one to describe the tracking workflows.

aphecetche committed:
Add TOC and reshuffle a bit
